task.go

```diff
@@ -296,8 +297,24 @@ func (e *Executor) RunTask(ctx context.Context, call taskfile.Call) error {
 		e.Logger.Errf(logger.Red, "task: cannot make directory %q: %v", t.Dir, err)
 	}
 
+	var firstTaskError error
 	for i := range t.Cmds {
-		if err := e.runCommand(ctx, t, call, i); err != nil {
+		cmd := t.Cmds[i]
+
+		// Once a task has encountered an error, only run tasks with always: true
+		if firstTaskError != nil && !cmd.Always {
+			continue
+		}
+
+		// Give commands with always: true their own context so that cleanup can run even after an interrupt signal
+		cmdCtx := ctx
+		if cmd.Always {
+			cleanupCtx, cancelCleanupCtx := context.WithTimeout(context.Background(), time.Second*10)
+			defer cancelCleanupCtx()
+			cmdCtx = cleanupCtx
+		}
```

Comment:
I'm not 100% sure whether this is the correct approach or even desired/necessary - completely open to changing or removing this code based on feedback